pkg/cluster/tinkerbell.go

```diff
@@ -125,6 +125,16 @@ func getTinkerbellMachineConfigs(ctx context.Context, client Client, c *Config)
 		}
 
 		c.TinkerbellMachineConfigs[machineConfig.Name] = machineConfig
+
+		if !machineConfig.Spec.TemplateRef.IsEmpty() {
+			c.TinkerbellTemplateConfigs = map[string]*anywherev1.TinkerbellTemplateConfig{}
```

Comment:
wont this get reset for each template config? So only the last TemplateRef will get set. We can have different templateRefs for each machineconfig right?

Reply:
oh yeah, good catch. Fixing it!